src/ui/public/courier/data_source/build_query/index.js

```diff
@@ -0,0 +1,33 @@
+import { groupBy } from 'lodash';
+import { DecorateQueryProvider } from '../_decorate_query';
+import { buildQueryFromKuery } from './from_kuery';
+import { buildQueryFromFilters } from './from_filters';
+import { buildQueryFromLucene } from './from_lucene';
+
+export function BuildESQueryProvider(Private) {
```

Comment:
There's probably no way around using `bool`, but maybe composing the results of the various contributing languages in a more restricted becomes necessary at some point. I can't come up with a more specific alternative, but maybe it becomes clear when we have a more complex kuery language and try mixing it with old filters. Just something to keep in mind, I guess.